pkg/pillar/diskmetrics/usage.go

```diff
@@ -14,9 +14,15 @@ import (
 // SizeFromDir performs a du -s
 func SizeFromDir(log *base.LogObject, dirname string) uint64 {
 	var totalUsed uint64
-	locations, err := ioutil.ReadDir(dirname)
+	f, err := os.Open(dirname)
 	if err != nil {
-		//log.Debugf("Dir %s is missing. Set the size to zero\n", dirname)
+		//log.Debugf("Exception while opening %s. %s. Set the size to zero\n", err.Error(), dirname)
+		return totalUsed
+	}
+	locations, err := f.Readdir(-1)
+	f.Close()
+	if err != nil {
+		//log.Debugf("Exception while reading dirs %s. Set the size to zero\n", err.Error(), dirname)
```

Comment:
Would it make sense to change this to a log.Errorf?
I don't know under what conditions it can fail.

Reply:
Changed.